core/cli/src/main/java/mil/nga/giat/geowave/core/cli/converters/PasswordConverter.java

```diff
@@ -0,0 +1,241 @@
+						if (propertyFilePath != null && !"".equals(propertyFilePath.trim())) {
+							propertyFilePath = propertyFilePath.trim();
+							File propsFile = new File(
+									propertyFilePath);
+							if (propsFile != null && propsFile.exists()) {
+								Properties properties = PropertiesUtils.fromFile(propsFile);
+								if (propertyKey != null && !"".equals(propertyKey.trim())) {
+									propertyKey = propertyKey.trim();
+								}
+								if (properties != null && properties.containsKey(propertyKey)) {
+									return decryptValue(properties.getProperty(propertyKey));
+								}
+							}
+							else {
+								try {
+									throw new ParameterException(
+											new FileNotFoundException(
+													propsFile != null ? "Properties file not found at path: "
+															+ propsFile.getCanonicalPath()
+															: "No properties file specified"));
+								}
+								catch (IOException e) {
+									throw new ParameterException(
+											e);
+								}
+							}
+						}
+						else {
+							throw new ParameterException(
+									"No properties file path specified");
+						}
+					}
+					else {
+						throw new ParameterException(
+								"Property File values are expected in input format <property file path>::<property key>");
+					}
+				}
+				else {
+					throw new ParameterException(
+							new Exception(
+									"No properties file specified"));
+				}
+				return value;
+			}
+		},
+		STDIN(
+				PasswordConverter.STDIN) {
+			private String input = null;
+
+			@Override
+			public boolean matches(
+					String value ) {
+				return prefix.equals(value);
+			}
+
+			@Override
+			String process(
+					String value ) {
+				if (input == null) {
+					input = promptAndReadPassword("Enter password: ");
+				}
+				return input;
+			}
+		},
+		DEFAULT(
+				"") {
+			@Override
+			String process(
+					String password ) {
+				return decryptValue(password);
+			}
+		};
+
+		String prefix;
+
+		private KeyType(
+				String prefix ) {
+			this.prefix = prefix;
+		}
+
+		public boolean matches(
+				String value ) {
+			return value.startsWith(prefix);
+		}
+
+		public String convert(
+				String value ) {
+			return process(value.substring(prefix.length()));
+		}
+
+		String process(
+				String value ) {
+			return value;
+		}
+	}
+
+	@Override
+	public String convert(
+			String value ) {
+		for (KeyType keyType : KeyType.values()) {
+			if (keyType.matches(value)) {
+				String convertedValue = keyType.convert(value);
+				// update the value in the configs properties file
+				if (updatePasswordInConfigs() && getPropertyKey() != null && !"".equals(getPropertyKey().trim())) {
+					Properties configProps = getGeoWaveConfigProperties();
+					if (configProps != null) {
+						// encrypt the value to be stored in the properties
+						convertedValue = encryptValue(convertedValue);
+						configProps.put(
+								getPropertyKey(),
+								convertedValue);
+						ConfigOptions.writeProperties(
```

Comment:
seems outside of the scope of a "converter" to write to the properties file here.